apps/05Mastermind.js

```diff
@@ -15,29 +15,73 @@ function printBoard() {
     }
 }
 
-function generateSolution() {
+function generateSolution(solution, guess) {
     for (var i = 0; i < 4; i++) {
         var randomIndex = getRandomInt(0, letters.length);
         solution += letters[randomIndex];
     }
 }
 
 function getRandomInt(min, max) {
-  return Math.floor(Math.random() * (max - min)) + min;
+    return Math.floor(Math.random() * (max - min)) + min;
 }
 
-function generateHint() {
+function generateHint(solution, guess) {
     // your code here
+
+    var solutionArray = solution.split('');
+    var guessArray = guess.split('');
+
+    var correctLetterLocations = 0;
+    var correctLetters = 0
+
+    for (var i = 0; i < solutionArray.length; i++) {
+        if (solutionArray[i] == guessArray[i]) {
+            correctLetterLocations++;
+            solutionArray[i] == null;
+        }
+    }
+
+    for (var i = 0; i < solutionArray.length; i++) {
```

Comment:
This is going to do the same thing as the loop counting correctLetterLocations. We just want to verify here if any elements from the solution array exist in the guest array. Remember we can use array.indexOf to see if something is in an array:

`
if(guessArray.indexOf(solutionArray[i]) != -1){
correctLetters++
}
`